lib/td/command/sched.rb

```diff
@@ -323,5 +326,75 @@ def sched_run(op)
     puts cmd_render_table(rows, :fields => [:JobID, :Time], :max_width=>500, :render_format => op.render_format)
   end
 
+  def sched_result(op)
+    options = job_show_options(op)
+
+    op          = options[:op]
+    verbose     = options[:verbose]
+    wait        = options[:wait]
+    output      = options[:output]
+    format      = options[:format]
+    render_opts = options[:render_opts]
+    limit       = options[:limit]
+    exclude     = options[:exclude]
+
+    back_number = 1
+    op.on('--last [Number]', "show the result before N from the last. default: 1") do |n|
+      back_number = n ?  n.to_i : 1
+    end
+
+    # save argv before calling cmd_parse, which removes flags from the argv array
+    argv_saved = op.argv.dup
+    name = op.cmd_parse
+
+    client = get_client
+    history = get_history(client, name, (back_number - 1), back_number)
+
+    job = history.first
+
+    if job.nil?
+      $stderr.puts "No jobs available for this query. Refer to 'sched:history'."
+      exit 1
+    end
+
+    # build the job:show command now
+    argv = job_show_option_argv(argv_saved, name, back_number)
+    argv << job.job_id.to_s
```

Comment:
Oops. client.history() returns job_id as an Integer? It should be fixed at API or td-client-ruby.